developers/advanced/null-value-support.md

```diff
@@ -1,20 +1,23 @@
 # Null value support
 
-For performance reasons, null handling support is disabled by default in Apache Pinot.
-When null support is disabled, all columns are treated as not null.
-Predicates like `IS NOT NULL` evaluates to `true,` and `IS NULL` evaluates to `false`.
-Aggregation functions like `COUNT`, `SUM`, `AVG`, `MODE`, etc. treat all columns as not null.
+{% hint style="info" %}
+**Multi-stage engine**
+
+This document begins with null handling for the [single-stage query engine](../../reference/single-stage-engine.md). For the [multi-stage query engine](../../reference/multi-stage-engine.md) (v2), see the [Enable basic null support for the multi-stage query engine](#enable-basic-null-support-for-the-multi-stage-query-engine) section.
+{% endhint %}
+
+Null handling is defined in two different parts: at ingestion and at query time.
+* [Basic null handling support](#basic-null-handling-support) means that you have enabled null handling at ingestion.
+* [Advanced null support](#advanced-null-handling-support) means that you have also enabled null handling at query time.
+
+## Basic null handling support
+
+By default, null handling is disabled (`nullHandlingEnabled=false`) in the Table index configuration ([tableIndexConfig](https://docs.pinot.apache.org/configuration-reference/table#tableindexconfig-1)). When null support is disabled, `IS NOT NULL` evaluates to `true,` and `IS NULL` evaluates to `false`. For example, the predicate in the query below matches all records.
```

Comment:
I don't think we should add this paragraph here. It has several issues:
1. It changes the layout. The table below summarizes the whole document, but with this paragraph it seems to only belong to `Basic null handling support`.
2. It mentions `nullHandlingEnabled=false`, which is
   * Too specific. Same things apply to column level nullability
   * Confusing. That syntax is very similar to the one used to enable advanced null handling.
3. Repetitive. The last sentence is repeated in the next paragraph.